presto-main/src/main/java/com/facebook/presto/sql/planner/plan/DistinctLimitNode.java

```diff
@@ -83,18 +85,16 @@ public boolean isPartial()
         return hashSymbol;
     }
 
+    @JsonProperty
     public List<Symbol> getDistinctSymbols()
     {
-        if (hashSymbol.isPresent()) {
-            return ImmutableList.copyOf(Iterables.filter(getOutputSymbols(), not(hashSymbol.get()::equals)));
-        }
-        return getOutputSymbols();
+        return distinctSymbols;
     }
 
     @Override
     public List<Symbol> getOutputSymbols()
     {
-        return source.getOutputSymbols();
+        return distinctSymbols;
```

Comment:
I think `DistinctLimitNode` produces pages with the same channels as source operator. Therefore output symbols should have the same order. Otherwise channels would be computed incorrectly for parent operator.